src/utils/querySelectorHelper.js

```diff
@@ -46,13 +47,19 @@ export function querySelectorAll(element, query) {
 /**
  * Function processes one element using current rule
  *
- * @param element
- * @param rules
+ * @param {HTMLElement} element - element to check
+ * @param {*} rules - current rules to check against element
+ * @param {boolean} skipSubChildren - flag if true, will not check subchildren of element
  * @return {boolean}
  */
-function* processElement(element, rules) {
+function* processElement(element, rules, skipSubChildren = false) {
```

Comment:
I think I would prefer `processElement` and `processRecursiveElement` or somthing like that over `skipSubChildren`